### DISCUSSION TITLE
Branched glycans and AllChem.EmbedMultipleConfs

### DISCUSSION THREAD

**Discussion by xavioist** [rdkit/rdkit#4623] [General] [answered]
Greetings,

I am currently working with glycans (generated using glycam carbohydrate builder @ glycam.org) and rdkit and I have found an issue regarding the EmbedMultipleConfs function.

The lineal glycans I'm able to load them from pdb and use EmbedMultipleConfs(), but with branched ones I get an empty list from the function. I have manually edited the pdb files, checked the hydrogen atoms, spacing and formatting, etc and the error still persists. I have tried many other different molecules and it has worked fine, only with branched glycans I've encountered this error. 

Find below the code I'm using to test this as well as the structures. 

Thank you very much for your help,

Xavi

**Code:**

#import rdkit necessary modules

my_mol = Chem.MolFromPDBFile(str(molecule_file), sanitize=True, removeHs=False)
my_mol.RemoveAllConformers()
cids = AllChem.EmbedMultipleConfs(my_mol, numConfs=100, maxAttempts=500)

[pdb_structures.zip](https://github.com/rdkit/rdkit/files/7383335/pdb_structures.zip)

**Comment by dehaenw** [comment, marked as answer]
issue #3323 might be helpful.
```python
my_mol = Chem.MolFromPDBFile("/tmp/Branched-medium-glycan-not-working.pdb", sanitize=True, removeHs=False)
my_mol.RemoveAllConformers()
cids = AllChem.EmbedMultipleConfs(my_mol,numConfs=100, maxAttempts=500,randomSeed=0,useRandomCoords=True)
print(len(cids))
```
gives
`100`
without the randomSeed and useRandomCoords arguments it gives
`0`

**Comment by xavioist** [comment]
Thank you very much for your quick answer! It was really helpful.

I wanted to add that your solution does generate conformers, but for the longer chains when I apply AllChem.GetConformerRMS() I get that the newly generated conformers are the same as the original molecule (I get a RMS of zero for all the cids)

**Code**:

rms_list = [AllChem.GetConformerRMS(mol, 0, i, prealigned=True) for i in range(0, len(cids))]
print(rms_list)

[0.0, 0.0, 0.0, 0.0, ... , 0.0]

This changes when switching the randomSeed from 0 to 1 (or other values), which gives me a lit of cids with RMS different from zero.

Thank you very much for helping me. Keep up the good work!

Xavi
